File: manipulacion-de-texto/split_strings.py
'''
/*
Complete la solución para que divida la cadena en cadenas de dos caracteres
en una lista o matriz (según el lenguaje que utilice).
 Si la cadena contiene un número impar de caracteres,
  debe reemplazar el segundo carácter faltante del par final con un guion bajo ('_').
*/
'''
def solution(s: str) -> list:
    if not s:
        return []
    new_list: list = []
    for i in range(0,len(s), 2):
        if i + 1 < len(s):
            new_list.append(s[i] + s[i + 1])
        else:
            new_list.append(s[i] + "_")
    return new_list

print(solution('abcdef'))
print(solution('abc'))

File: manipulacion-de-texto/split_strings.dart
/*
Complete la solución para que divida la cadena en cadenas de dos caracteres
en una lista o matriz (según el lenguaje que utilice).
 Si la cadena contiene un número impar de caracteres,
  debe reemplazar el segundo carácter faltante del par final con un guion bajo ('_').
*/
void main() {
  print(solution('abcdef'));
  print(solution('abc'));
}

List<String> solution(String s) {
  if (s.isEmpty) {
    return [];
  }
  List<String> new_list = [];
  for (int i = 0; i < s.length; i += 2) {
    if (i + 1 < s.length) {
      new_list.add(s[i] + s[i + 1]);
    } else {
      new_list.add(s[i] + "_");
    }
  }
  return new_list;
}
